src/Microsoft.AspNetCore.Mvc.Razor/Internal/CSharpCompiler.cs

```diff
@@ -154,9 +154,7 @@ private void EnsureOptions()
 
             var parseOptions = new CSharpParseOptions(preprocessorSymbols: defines);
 
-            LanguageVersion languageVersion;
-            if (!string.IsNullOrEmpty(dependencyContextOptions.LanguageVersion) &&
-                Enum.TryParse(dependencyContextOptions.LanguageVersion, ignoreCase: true, result: out languageVersion))
+            if (LanguageVersionFacts.TryParse(dependencyContextOptions.LanguageVersion, out var languageVersion))
             {
                 parseOptions = parseOptions.WithLanguageVersion(languageVersion);
             }
```

Comment:
Add an empty else with a comment (or debug assert) stating that failure to parse the lang version can't really happen and why.